eth_tester/backends/pyevm/main.py

```diff
@@ -249,11 +249,11 @@ def take_snapshot(self):
     def revert_to_snapshot(self, snapshot):
         block = self.chain.get_block_by_hash(snapshot)
         if block.number > 0:
-            self.chain.chaindb.set_as_canonical_chain_head(block.header)
+            self.chain.chaindb._set_as_canonical_chain_head(block.header)
```

Comment:
ewww, do we no longer have a public API we can access for this?  We should fix that (not now but soon) by moving it back to being a public API.

cc @carver to ask what the reason was for making this private?

Reply:
In some ways, it was to highlight issues like this. The public call should be to [`persist_header_to_db()`](https://github.com/ethereum/py-evm/blob/master/evm/db/chain.py#L262). Giving callers a way to reach into the EVM and violate its own fork choice rule seems like asking for trouble.

In this case, it seems like we should actually be using py-evm's own `snapshot()` and `revert()` here, anyway.

Reply:
Hrm, off the top of my head, the py-evm `snapshot()` and `revert()` methods only work within block boundaries.  Need to go recheck this.  If that's the case we need to come up with an alternate solution.